Mathlib/NumberTheory/LSeries/Dirichlet.lean

```diff
@@ -0,0 +1,420 @@
+  · by_contra! h
+    have h' : s.re ≤ (1 : ℂ).re := by simp only [one_re, h]
+    exact not_LSeriesSummable_moebius_at_one <| LSeriesSummable.of_re_le_re h' H
+  · rw [abs_intCast, ← Int.cast_abs]
+    norm_cast
+    rcases eq_or_ne (μ n) 0 with h | h
+    · simp [h]
+    · rcases moebius_ne_zero_iff_eq_or.mp h with h | h <;> simp [h]
+
+/-- The abscissa of absolute convergence of the L-series of the Möbius function is `1`. -/
+lemma abscissaOfAbsConv_moebius : abscissaOfAbsConv ↗μ = 1 := by
+  simpa only [abscissaOfAbsConv, LSeriesSummable_moebius_iff, ofReal_re, Set.Ioi_def,
+    EReal.image_coe_Ioi, EReal.coe_one] using csInf_Ioo <| EReal.coe_lt_top _
+
+end ArithmeticFunction
+
+end Moebius
+
+/-!
+### L-series of Dirichlet characters
+-/
+
+lemma LSeries.one_convolution_eq_zeta_convolution {R : Type*} [Semiring R] (f : ℕ → R):
+    (1 : ℕ → R) ⍟ f = ((ArithmeticFunction.zeta ·) : ℕ → R) ⍟ f := by
+  ext ⟨- | n⟩
+  · simp only [Nat.zero_eq, convolution_map_zero, ArithmeticFunction.zeta_apply, Nat.cast_ite,
+      Nat.cast_zero, Nat.cast_one]
+  · simp only [convolution_def, Pi.one_apply, one_mul, ArithmeticFunction.zeta_apply, Nat.cast_ite,
+      Nat.cast_zero, Nat.cast_one, ite_mul, zero_mul]
+    refine Finset.sum_congr rfl fun p hp ↦ ?_
+    simp only [(Nat.ne_zero_of_mem_divisorsAntidiagonal hp).1, ↓reduceIte]
+
+/-- `χ₁` is (local) notation for the (necessarily trivial) Dirichlet charcater modulo `1`. -/
+local notation (name := Dchar_one) "χ₁" => (1 : DirichletCharacter ℂ 1)
+
+namespace DirichletCharacter
+
+open LSeries Nat Complex
+
+/-- Twisting by a Dirichlet character `χ` distributes over convolution. -/
+lemma mul_convolution_distrib {R : Type*} [CommSemiring R] {n : ℕ} (χ : DirichletCharacter R n)
+    (f g : ℕ → R) :
+    ((χ ·) : ℕ → R) * (f ⍟ g) = (((χ ·) : ℕ → R) * f) ⍟ (((χ ·) : ℕ → R) * g) := by
+  ext n
+  simp only [Pi.mul_apply, LSeries.convolution_def, Finset.mul_sum]
+  refine Finset.sum_congr rfl fun p hp ↦ ?_
+  rw [(Nat.mem_divisorsAntidiagonal.mp hp).1.symm, Nat.cast_mul, map_mul]
+  exact mul_mul_mul_comm ..
+
+open ArithmeticFunction in
+/-- The convolution of a Dirichlet character `χ` with the twist `χ * μ` is `δ`,
+the indicator function of `{1}`. -/
+lemma convolution_mul_moebius {n : ℕ} (χ : DirichletCharacter ℂ n) : ↗χ ⍟ (↗χ * ↗μ) = δ := by
+  have : (1 : ℕ → ℂ) ⍟ (μ ·) = δ := by
+    rw [one_convolution_eq_zeta_convolution, ← one_eq_delta]
+    change ⇑(ζ : ArithmeticFunction ℂ) ⍟ ⇑(μ : ArithmeticFunction ℂ) = ⇑(1 : ArithmeticFunction ℂ)
+    simp only [coe_mul, coe_zeta_mul_coe_moebius]
+  nth_rewrite 1 [← mul_one ↗χ]
+  simpa only [← mul_convolution_distrib χ 1 ↗μ, this] using
+    mul_delta <| by simp only [cast_one, map_one]
+
+lemma mul_delta {n : ℕ} (χ : DirichletCharacter ℂ n) : ↗χ * δ = δ :=
+  LSeries.mul_delta <| by rw [cast_one, MulChar.map_one]
+
+lemma delta_mul {n : ℕ} (χ : DirichletCharacter ℂ n) : δ * ↗χ = δ :=
+  mul_comm δ _ ▸ mul_delta ..
```

Comment:
If you move `mul_delta` up, you can use it in the previous lemma

Reply:
OK; it simplifies the last line in the proof a bit.